client/client.go

```diff
@@ -142,6 +146,20 @@ func NewClientWithOpts(ops ...func(*Client) error) (*Client, error) {
 			c.scheme = "https"
 		}
 	}
+	sc, err := stackClient.NewClientWithSettings(stackClient.Settings{
+		Scheme:            c.scheme,
+		Host:              c.host,
+		Proto:             c.proto,
+		Addr:              c.addr,
+		BasePath:          c.basePath,
+		Client:            c.client,
+		Version:           c.version,
+		CustomHTTPHeaders: c.customHTTPHeaders,
+	})
```

Comment:
I should point out that we're not 100% happy with this implementation detail just yet, but what we've been aiming for is a pattern where projects integrating into Moby can define their own API and client libraries that can work in isolation, and also be integrated into the overall client library with minimal integration glue.  There's more cut-and-paste code in https://github.com/docker/stacks/tree/master/pkg/client than we'd like, so ultimately where I think this should go is more of the APIs docker/docker/client are teased out into importable pkg modules that are reusable.

Reply:
What are you missing? from the client package here?

Reply:
Should this be passed as an `op` perhaps?

```go
NewClientWithOpts(client.WithStackClient(...))
```

Reply:
These were the two major chunks
* https://github.com/docker/stacks/blob/master/pkg/client/errors.go
* https://github.com/docker/stacks/blob/master/pkg/client/request.go

Reply:
I'm really not sure what this subclient is for.
We have the client in github.com/docker/docker/client -- what is missing from the client for you to implement a stacks client? Are you looking for a more HTTP client wrapper?

Reply:
For the errors, there's some changes in https://github.com/moby/moby/pull/38467 (unifies some of those with the errdefs package)

Reply:
> I'm really not sure what this subclient is for

It's a client library for dealing with stacks

https://github.com/docker/stacks/blob/master/pkg/client/interface.go#L10-L17

Reply:
> what is missing from the client for you to implement a stacks client?

We had to copy-and-paste a bit of the low-level code for interacting with a Docker REST style API server, since the functions in the docker/docker/client package were not exported.  As we try to break up the monolith and fulfill the Moby vision, other projects will wind up re-inventing this same low-level http client code, so it seems like we should consider generalizing that so its reusable.

Reply:
Seems like these could be refactored into a HTTP client wrapper.

Reply:
@cpuguy83 in general, what are you thinking by "refactored into a HTTP client wrapper"? i'm unclear on what you mean by that.

Reply:
I mean make a new package for the low level HTTP handling that can be shared by both github.com/docker/docker/client and github.com/docker/stacks/pkg/client.